### PR TITLE
Feature/cicd deploy

### PR DESCRIPTION
**Pull request by jonathansmolenski**
This PR contains the CI/CD code for deploying to environments based on pushes to certain branches.

Currently there is a bug with the terraform apply regarding an SSM parameter that @joshgarde is aware of.

As with the other swodlr deployments, there is currently no smoke test to validate that things have stood up properly.  I'd like to add one later.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -62,37 +81,79 @@ jobs:
...
+        if: github.ref == 'refs/heads/main' && github.event_name != 'workflow_dispatch'
+        run: |
+          bumpver --tag final
       #   -- Build --
       - name: Build lambda package
-        run: ./build.sh
+        id: package-build
+        run: |
+          ./build.sh
+          echo "deploy_env=${{ env.TARGET_ENV }}" >> $GITHUB_OUTPUT
```
Comment (joshgarde) [inline .github/workflows/build.yml]: Not sure why this needs to be run here as part of "Build lambda package"

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -62,37 +81,79 @@ jobs:
...
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: us-west-2
+          role-session-name: GitHubActions
+          aws-access-key-id: ${{ secrets[vars.AWS_ACCESS_KEY_ID_SECRET_NAME] }}
+          aws-secret-access-key: ${{ secrets[vars.AWS_SECRET_ACCESS_KEY_SECRET_NAME] }}
+          mask-aws-account-id: true
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{ github.repository }}
```
Comment (joshgarde) [inline .github/workflows/build.yml:137]: Is it required to specify this? I thought by default checkout will checkout the current repo the workflow is being run in

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -62,37 +81,79 @@ jobs:
...
           TF_VAR_sds_username:     ${{ secrets.SDS_USERNAME }}
           TF_VAR_sds_password:     ${{ secrets.SDS_PASSWORD }}
         run: |
           echo "${{ secrets.JPLICA_CERT }}" >> ${{ runner.temp }}/JPLICA.pem
- 
-          source bin/config.sh sit
-          terraform apply -auto-approve
+
+          source bin/config.sh ${{ vars.TF_VENUE }}
+          terraform apply -auto-approve
```
Comment (joshgarde) [inline .github/workflows/build.yml]: Newline missing

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -62,37 +81,79 @@ jobs:
...
-          AWS_DEFAULT_REGION:      us-west-2
-
+          AWS_DEFAULT_REGION: us-west-2
+          TF_VAR_edl_base_url: ${{ secrets.EDL_BASE_URL }}
+          TF_VAR_edl_client_id: ${{ secrets.EDL_CLIENT_ID }}
+          TF_VAR_edl_client_secret: ${{ secrets.EDL_CLIENT_SECRET }}
+          TF_VAR_session_encryption_key: ${{ secrets.SESSION_ENCRYPTION_KEY }}
+          TF_VAR_ingest_aws_account: ${{ secrets.INGEST_AWS_ACCOUNT }}
+          TF_VAR_ingest_aws_role: ${{ secrets.INGEST_AWS_ROLE }}
+          ## Original deployment values
```
Comment (joshgarde) [inline .github/workflows/build.yml]: No need to demarcate the two

**Review by joshgarde — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (jonathansmolenski) [reply .github/workflows/build.yml:137]: That's just sort of the standard way it's always shown, but yeah basically this is explicitly setting the default value.  We can remove it across all repos if you like

**Pull request merged**